### PR TITLE
MP-3270 iOS/Android Events Integrations

### PR DESCRIPTION
**Pull request by sahil-rokt**
### Background ###

This PR is to integrate iOS partner Events.

Fixes [([issue](https://rokt.atlassian.net/browse/MP-3270))]

### What Has Changed: ###

List out what has changed as a result of this PR.

### How Has This Been Tested? ###

Screenshot 2024-04-11 at 2 48 19 PM

Describe the tests that you ran to verify your changes. 

### Notes

Add any notes or extra information here that might be useful to the reviewer if applicable i.e. links to documentation/blogs or dashboards.

### Checklist: ###

- [x] I have performed a self-review of my own code.
- [x] I have made corresponding changes to the documentation.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [x] I have verified that CI completes successfully.
- [x] All insignificant commits have been squashed.

### CODE REVIEW

**Comment by danial-rokt** [comment]
better to merge to release-4.3.x branch?

**Comment by sahil-rokt** [comment]
> better to merge to release-4.3.x branch?

:) thanks

**Review by danial-rokt — APPROVED** [review]

**Review by mirannakamura — APPROVED** [review]

**Review by thomson-t — APPROVED** [review]

File: ios/Classes/RoktMethodCallHandler.swift [hunk 1]
```diff
@@ -63,28 +83,71 @@ struct RoktMethodCallHandler {
             let callBackId = args["callbackId"] as? Int ?? 0
             var callbackMap = [String: Any] ()
             callbackMap["id"] = callBackId
+            Rokt.events(viewName: viewName, onEvent: { roktEvent in
+                var eventParamMap = [String: String] ()
+                var eventName = ""
+                var placementId: String?
+                if let event = roktEvent as? RoktEvent.FirstPositiveEngagement {
+                    eventName = "FirstPositiveEngagement"
+                    placementId = event.placementId
```
Comment (thomson-t) [inline ios/Classes/RoktMethodCallHandler.swift:92]: Handle the fulfillment here as well?

File: android/src/main/kotlin/com/rokt/rokt_sdk/MethodCallHandlerImpl.kt [hunk 2]
```diff
@@ -97,6 +108,60 @@ class MethodCallHandlerImpl(
...
+                    val params: MutableMap<String, String> = mutableMapOf()
+                    params["viewName"] = viewName
+                    val placementId: String? = when (event) {
+                        is RoktEvent.HideLoadingIndicator -> {
+                            params["event"] = "HideLoadingIndicator"
+                            null
+                        }
+                        is RoktEvent.FirstPositiveEngagement -> {
+                            params["event"] = "FirstPositiveEngagement"
+                            event.id
```
[the remark this answers is not in the document]
Comment (thomson-t) [reply android/src/main/kotlin/com/rokt/rokt_sdk/MethodCallHandlerImpl.kt:123]: Something like this https://github.com/ROKT/rokt-sdk-react-native/pull/132/files#diff-7bff632860b19d711f0150be402c1c83ba09def3a50ccce30953d9122afae6b4R227

File: ios/Classes/RoktMethodCallHandler.swift (hunk 1 above)
Comment (thomson-t) [reply ios/Classes/RoktMethodCallHandler.swift:92]: https://github.com/ROKT/rokt-sdk-react-native/pull/132/files#diff-abb31688af2dfba056baf7e09192b94bcd5072727d39c0c4d6abc5c6d6e2100cR102

File: android/src/main/kotlin/com/rokt/rokt_sdk/MethodCallHandlerImpl.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (sahil-rokt) [reply android/src/main/kotlin/com/rokt/rokt_sdk/MethodCallHandlerImpl.kt:123]: As discussed, Flutter does not support `FulfilmentAttribue` or `execute2Step` and we would need a generic api to receive the event data from Partner App which will be feature extension.

**Review by thomson-t — APPROVED** [review]

**Pull request merged**